registration/views.py

```diff
@@ -46,5 +51,50 @@ def _register(request, FormSet, args):
 
     args.update(csrf(request))
     args.update({'form' : form })
-    args.update({'formset': formset })  
-    return render(request, 'registration/register.html', args)
+    args.update({'formset': formset })
+    args.update({'troop_modal_form': troop_modal_form})
+    args.update({'council_modal_form': council_modal_form})
+    args.update({'troop_modal_title': "Add Troop"})
+    args.update({'council_modal_title': "Add Council"})
+    return render(request, 'registration/register_user.html', args)
+
+def register(request):
+    return render(request, 'registration/register.html')
+
+def register_venture(request):
+    args = {'title': 'Register Venture'}
+    ct = ContentType.objects.get_for_model(Venture)
+    #p = Permission.objects.get(content_type=ct, codename='can_modify_venture_enrollments')
+    #args.update({'perms':p})
+    FormSet = VentureFormSet
+    return _register(request, FormSet, args)
+
+def register_troopcontact(request):
+    args = {'title': 'Register TroopContact'}
+    ct = ContentType.objects.get_for_model(TroopContact)
+    FormSet = TroopContactFormSet
+    return _register(request, FormSet, args)
+
+def register_volunteer(request):
+    args = {'title': 'Register Volunteer'}
+    ct = ContentType.objects.get_for_model(Volunteer)
+    FormSet = VolunteerFormSet
+    return _register(request, FormSet, args)
+
+def register_troop(request):
+    if request.POST:
+        form = TroopForm(request.POST)
+        if form.is_valid():
+            form.save()
+            latest_troop = Troop.objects.order_by('-pk')[:1]
+            data = serializers.serialize("json", latest_troop)
+            return HttpResponse(JsonResponse(data, safe=False))
+
+def register_council(request):
+    if request.POST:
+        form = CouncilForm(request.POST)
+        if form.is_valid():
+            form.save()
+            latest_council = Council.objects.order_by('-pk')[:1]
+            data = serializers.serialize("json", latest_council)
+            return HttpResponse(JsonResponse(data, safe=False))
```

Comment:
Without trying it -- I think you can get rid of the extra JSON.parse() call by returning the JsonResponse() without wrapping it in an HttpResponse (JsonResponse is a subclass of HttpResponse).  Otherwise, get rid of the JsonResponse completely and use HttpResponse and set content-type to "application/json".